Dockerfile

```
@@ -0,0 +1,12 @@
+FROM node:15.14-alpine AS build
+WORKDIR /src
+COPY . .
+RUN npm ci
+RUN npm run build
+
+FROM node:15.14-alpine as final
+RUN yarn global add serve
```

why do we use `yarn` here instead of `npm`?

---

This whole setup could happen by running only `npx serve` instead. I don't see justified this additional layer.

---

I'm following the readme instructions. Will try to use `npx serve`

---

> The quickest way to get started is to just run npx serve in your project's directory.
https://www.npmjs.com/package/serve#usage

This is the first sentence in the readme instructions, after that just an other way is explained how to do it.

---

Running with `npx serve` it says:
`npm WARN exec The following package was not found and will be installed: serve`

So it's to choose if install it with container, or with first execution. My opinion, best to include before. A container should work as is, minimizing requirements to external resources to run. More predictable.

---

yeah I see now, you have only the `build` folder. you are right, I just wanted to propose to use only one package manager in the same workflow, can you try with this please?
```suggestion
RUN npm i -g serve
```

---

Thanks @tmm360 and @nugaon , we just merged a PR which adds serve as devDependency so that users who want to use it directly outside of Docker have simpler life. I believe it also solves this discussion. Cheers :) https://github.com/ethersphere/bee-dashboard/pull/79

---

@vojtechsimetka Unfortunately because `serve` is imported as a dev dependency, and I'm optimizing image exporting only build result, serve isn't available on the final image.
Instead "RUN npm i -g serve" works. At this point I'm going to take this last one.